src/components/Inventory/InventoryCreate.tsx

```diff
@@ -42,8 +37,46 @@ export default function CreateInventory(): JSX.Element {
   const history = useHistory();
   const snackbar = useSnackbar();
   const [validateFields, setValidateFields] = useState<boolean>(false);
-  const [record, setRecord, onChange] = useForm<CreateBatchRequestPayload>(defaultBatch());
   const [totalQuantity, setTotalQuantity] = useState(0);
+  const { selectedOrganization } = useOrganization();
+  const timeZoneFeatureEnabled = isEnabled('Timezones');
+  const [selectedNursery, setSelectedNursery] = useState<Facility>();
+  const tz = useLocationTimeZone().get(timeZoneFeatureEnabled ? selectedNursery : undefined);
+  const [timeZone, setTimeZone] = useState(tz.id);
+
+  const defaultBatch = (): CreateBatchRequestPayload =>
+    ({
+      addedDate: getTodaysDateFormatted(timeZone),
+      facilityId: undefined,
+      speciesId: undefined,
+      germinatingQuantity: undefined,
+      notReadyQuantity: undefined,
+      readyQuantity: undefined,
+    } as unknown as CreateBatchRequestPayload);
+
+  const [record, setRecord, onChange] = useForm<CreateBatchRequestPayload>(defaultBatch());
+
+  useEffect(() => {
+    if (record.facilityId) {
+      const batchNursery = getNurseryById(selectedOrganization, record.facilityId);
+      setSelectedNursery(batchNursery);
+    }
+  }, [record.facilityId, selectedOrganization]);
+
+  useEffect(() => {
+    if (timeZone !== tz.id) {
+      setTimeZone(tz.id);
+    }
+  }, [tz.id, timeZone]);
+
+  useEffect(() => {
+    setRecord((previousRecord: CreateBatchRequestPayload): CreateBatchRequestPayload => {
+      return {
+        ...previousRecord,
+        addedDate: getTodaysDateFormatted(timeZone),
```

Comment:
if the record has a date set by user, we should use that.. not sure how we can detect it here.. maybe this will be called before user edits?

Reply:
This is called everytime the user changes the "receiving nursery", so if the user selected the date before the nursery, when the nursery is changed, this date will be changed to today's date (in the time zone of the nursery)